src/adzerk/boot_reload.clj

```diff
@@ -21,10 +23,10 @@
          (sort-by :dependency-order)
          (map tmp-path))))
 
-(defn- start-server [pod {:keys [ip port] :as opts}]
+(defn- start-server [pod {:keys [ip port] :as opts} domain]
   (let [{:keys [ip port]}
         (pod/with-call-in pod (adzerk.boot-reload.server/start ~opts))
-        host (if-not (= ip "0.0.0.0") ip "localhost")]
+        host (if-not (string/blank? domain) domain (if-not (= ip "0.0.0.0") ip "localhost"))]
```

Comment:
Should be enough to test if domain is non-nil, and instead of two nested ifs, cond would probably be clearer:

```clj
(cond
  domain domain
  (= ip "0.0.0.0") "localhost"
  ip)
```